### PR TITLE
Clean parameter shutdown on scheduler crash

### LINKED ISSUE #229: Bug: Parameter Server not stopping
## Description

The PS can't be stopped with ctrl+c.

## Symptoms

```log
2025-12-11T12:34:03.209430Z  INFO hypha_worker::executor::parameter_server: Executing parameter server job job_spec=JobSpec { job_id: aa6f9913-88cf-41ca-ab99-6dac8e15ee8e, executor: Aggregate(AggregateExecutor { descriptor: AggregateExecutorDescriptor { name: "parameter-server" }, config: AggregateExecutorConfig { optimizer: Nesterov { learning_rate: 0.7, momentum: 0.9 } } }) }
2025-12-11T12:34:03.209807Z  INFO hypha_worker::arbiter: Task successfully dispatched job_id=f9fe7b34-879e-4506-aa3a-dd6784df2b86 scheduler_id=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce
^C2025-12-11T12:35:15.162850Z  INFO hypha_worker: Received SIGINT, shutting down
2025-12-11T12:35:19.918926Z  INFO hypha_worker::arbiter: Dropped 1 expired leases expired=[Lease { id: 47afcfa8-8fee-47e4-9148-1df64ce1c0f3, leasable: ResourceLease { peer_id: PeerId("12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce"), reservation: 5c4b7929-cccc-4d48-b6b0-9633bbc2a1a8 }, timeout: SystemTime { tv_sec: 1765456519, tv_nsec: 912519000 } }]
^C2025-12-11T12:35:22.359005Z ERROR hypha_network::request_response: Outbound request-response failure peer=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce request_id=OutboundRequestId(137) error=DialFailure
2025-12-11T12:35:22.498829Z ERROR hypha_network::request_response: Outbound request-response failure peer=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce request_id=OutboundRequestId(138) error=DialFailure
2025-12-11T12:35:23.391545Z ERROR hypha_network::request_response: Outbound request-response failure peer=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce request_id=OutboundRequestId(139) error=DialFailure
^C^C^C
```

This is a flaky error.

Feel free to share whatever helps us see what you see! 

IMPORTANT: Remember Github issues are public, so please do not include any sensitive information!
-->

### PR DESCRIPTION
**Pull request by nfnt**
If a scheduler is unreachable while a parameter server is trying to communicate with it, we would no longer be able to gracefully stop the parameter server executor. This has been resolved to ensure that cancellation tokens are respected when communicating with schedulers.

Remarks: We should revisit and refactor how we use cancellation tokens in executors. The current code has many different code paths that separately handle cancellation tokens, making it prone to bugs. This PR doesn't resolve that and further work is needed here.

Resolves #229

### CODE REVIEW

**Review by l45k — CHANGES_REQUESTED** [review]
This looks good. Maybe we should do the same in Line 324?

**Comment by nfnt** [comment]
`broadcast_update`, called in line 328 uses a cancellation token, so this code already handles cancellation.

**Review by l45k — APPROVED** [review]
👍
